transformer/external/executabletransformer.go

```diff
@@ -138,7 +165,24 @@ func (t *Executable) Transform(newArtifacts []transformertypes.Artifact, already
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to upload the transform input into the environment at the path '%s' . Error: %w", transformInputFile, err)
 	}
-	stdout, stderr, exitcode, err := t.Env.Exec(append(t.ExecConfig.TransformCMD, filepath.Join(containerInputDir, transformInputFile)))
+	t.ExecConfig.TransformScriptIO.Input.Path = filepath.Join(containerInputDir, transformInputFile)
+	t.ExecConfig.TransformScriptIO.Output.Path = filepath.Join(transformContainerOutputDir, transformOutputFile)
+	cmdToRun := t.ExecConfig.TransformCMD
+	if t.ExecConfig.TransformScriptIO.Input.Mode == argMode {
+		cmdToRun = append(cmdToRun, "input="+t.ExecConfig.TransformScriptIO.Input.Path)
+	} else if t.ExecConfig.TransformScriptIO.Input.Mode == envMode {
+		t.Env.EnvKeyValueList = append(t.Env.EnvKeyValueList,
+			t.ExecConfig.TransformScriptIO.Input.EnvName+envDelimiter+t.ExecConfig.TransformScriptIO.Input.Path)
+		t.Env.Env.AddEnvironmentVariablesToInstance([]string{t.ExecConfig.TransformScriptIO.Input.EnvName + envDelimiter + t.ExecConfig.TransformScriptIO.Input.Path})
+	}
+	if t.ExecConfig.TransformScriptIO.Output.Mode == argMode {
+		cmdToRun = append(cmdToRun, "output="+t.ExecConfig.TransformScriptIO.Output.Path)
+	} else if t.ExecConfig.TransformScriptIO.Output.Mode == envMode {
+		t.Env.EnvKeyValueList = append(t.Env.EnvKeyValueList,
+			t.ExecConfig.TransformScriptIO.Output.EnvName+envDelimiter+t.ExecConfig.TransformScriptIO.Output.Path)
+		t.Env.Env.AddEnvironmentVariablesToInstance([]string{t.ExecConfig.TransformScriptIO.Output.EnvName + envDelimiter + t.ExecConfig.TransformScriptIO.Output.Path})
+	}
+	stdout, stderr, exitcode, err := t.Env.Exec(cmdToRun)
```

Comment:
since the processing for transform and detect commands are same, might be better to put the processing in a function, and call here.

Reply:
done